tools/bridge/bridge/confirmation_task_planner.py

```diff
@@ -1,161 +1,88 @@
+
+from bridge.constants import (
+    COMPLETION_EVENT_NAME,
+    CONFIRMATION_EVENT_NAME,
+    TRANSFER_EVENT_NAME,
+)
+
+
+class TransferRecorder:
+    def __init__(self, sync_persistence_time: float) -> None:
+        self.sync_persistence_time = sync_persistence_time
+
+        self.transfer_events: Dict[Hash32, AttributeDict] = {}
+
+        self.transfer_hashes: Set[Hash32] = set()
+        self.confirmation_hashes: Set[Hash32] = set()
+        self.completion_hashes: Set[Hash32] = set()
+
+        self.scheduled_hashes: Set[Hash32] = set()
+
+        self.confirmations_synced_until = 0.0
+        self.completions_synced_until = 0.0
+
+    def apply_sync_completed(self, event: str, timestamp: float) -> None:
+        if event == TRANSFER_EVENT_NAME:
+            pass
+        elif event == CONFIRMATION_EVENT_NAME:
+            if timestamp < self.confirmations_synced_until:
+                raise ValueError("Sync time must never decrease")
+            self.confirmations_synced_until = timestamp
+        elif event == COMPLETION_EVENT_NAME:
+            if timestamp < self.completions_synced_until:
+                raise ValueError("Sync time must never decrease")
+            self.completions_synced_until = timestamp
         else:
-            raise ValueError("Got unknown event {event}")
-
-        recorder.record_transfer_hash(timestamp, transfer_hash)
-
-    def get_next_transfer_hashes(self) -> List[Hash32]:
-        transfer_time = self.transfer_event_recorder.current_timestamp
-        confirmation_time = self.confirmation_event_recorder.current_timestamp
-        completion_time = self.completion_event_recorder.current_timestamp
-
-        home_time = min(confirmation_time, completion_time)
-
-        start_time = self.transfers_processed_until
-        end_time = home_time - self.home_look_ahead
+            raise ValueError(f"Got unknown event {event}")
+
+    def apply_event(self, event: AttributeDict) -> None:
+        event_name = event.event
+        transfer_hash = Hash32(decode_hex(event.args.transferHash))
+        assert len(transfer_hash) == 32
+
+        if event_name == TRANSFER_EVENT_NAME:
+            self.transfer_hashes.add(transfer_hash)
+            self.transfer_events[transfer_hash] = event
+        elif event_name == CONFIRMATION_EVENT_NAME:
+            self.confirmation_hashes.add(transfer_hash)
+        elif event_name == COMPLETION_EVENT_NAME:
+            self.completion_hashes.add(transfer_hash)
+        else:
+            raise ValueError(f"Got unknown event {event}")
 
-        transfer_hashes = self.transfer_event_recorder.get_transfer_hashes(
-            start_time=start_time, end_time=end_time
+    def is_in_sync(self, current_time: float) -> bool:
+        synced_until = min(
+            self.confirmations_synced_until, self.completions_synced_until
         )
-        self.transfers_processed_until = transfer_time
+        return current_time <= synced_until + self.sync_persistence_time
```

Comment:
Probably we should make sure that `self.sync_persistence_time` is not negative on initialization?